java/vector/src/main/java/org/apache/arrow/vector/util/DecimalUtility.java

```diff
@@ -120,34 +121,25 @@ public static void writeLongToArrowBuf(long value, ArrowBuf bytebuf, int index)
    * width.
    */
   public static void writeByteArrayToArrowBuf(byte[] bytes, ArrowBuf bytebuf, int index) {
-    writeByteArrayToArrowBuf(bytes, bytebuf, index, 0);
+    writeByteArrayToArrowBufHelper(bytes, bytebuf, index);
   }
 
-  private static void writeByteArrayToArrowBuf(byte[] bytes, ArrowBuf bytebuf, int index, int padValue) {
+  private static void writeByteArrayToArrowBufHelper(byte[] bytes, ArrowBuf bytebuf, int index) {
     final int startIndex = index * DECIMAL_BYTE_LENGTH;
     if (bytes.length > DECIMAL_BYTE_LENGTH) {
       throw new UnsupportedOperationException("Decimal size greater than 16 bytes");
     }
 
     // Decimal stored as little endian, need to swap data bytes before writing to ArrowBuf
     byte[] bytesLE = new byte[bytes.length];
-    int stop = bytes.length / 2;
-    for (int i = 0, j; i < stop; i++) {
-      j = (bytes.length - 1) - i;
-      bytesLE[i] = bytes[j];
-      bytesLE[j] = bytes[i];
-    }
-    if (bytes.length % 2 != 0) {
-      int i = (bytes.length / 2);
-      bytesLE[i] = bytes[i];
+    int leStartIndex = bytes.length - 1;
+    for (int i = 0, j; i < bytes.length; i++, leStartIndex--) {
+      bytesLE[leStartIndex] = bytes[i];
```

Comment:
Simply write bytesLE[i] = bytes[bytes.length - 1 - i], and remove the leStartIndex variable? 
I think this will make the code clearer.

Reply:
sure fixed.